### PR TITLE
refactor: refactor general

### PR DESCRIPTION
**Pull request by ltfschoen**
@maciejhirsz how do you use the other functions mentioned in crates/kobold/js/utils.js?

for example `start` is in crates/kobold/js/utils.js
and also in crates/kobold/src/lib.rs
and then it's used in various examples main.rs files with `kobold::start(html! { <Elapsed />})`

so if i wanted to use `fragment` in crates/kobold/js/utils.js, is the idea just to add it to crates/kobold/src/lib.rs or in its own file in that folder, for example for `fragement` it'd just be
```rust
pub fn fragment() {
    init_panic_hook();

    util::__kobold_fragment();
}
```

before we can use it in an example main.rs file with `kobold::fragment()`?

but i can't see the purpose of `fragement`

### CODE REVIEW

File: crates/kobold/js/util.js [hunk 1]
```diff
@@ -7,12 +7,7 @@ export function __kobold_before(n,i) { n.before(i); }
 export function __kobold_unmount(n) { n.remove(); }
 export function __kobold_replace(o,n) { o.replaceWith(n); }
 export function __kobold_empty_node() { return document.createTextNode(""); }
-export function __kobold_fragment()
-{
-	let f = document.createDocumentFragment();
-	f.append("", "");
-	return f;
-};
+export function __kobold_fragment() { return document.createDocumentFragment().append("", ""); };
```
Comment (maciejhirsz) [inline crates/kobold/js/util.js:10]: `append` returns `undefined` so this change bricks this function

File: crates/kobold/js/util.js [hunk 2]
```diff
@@ -22,22 +17,16 @@ export function __kobold_fragment_unmount(f)
 {
 	let [b, e] = fragmentDecorators.get(f);
 	while (b.nextSibling !== e) f.appendChild(b.nextSibling);
-	f.appendChild(e);
-	f.insertBefore(b, f.firstChild);
+	f.appendChild(e).insertBefore(b, f.firstChild);
```
Comment (maciejhirsz) [inline crates/kobold/js/util.js:20]: `appendChild` returns the child appended, not the parent, this is now equivalent to:

```js
f.appendChild(e);
e.insertBefore(b, f.firstChild);
```
Which will also break

File: crates/kobold/js/util.js [hunk 3]
```diff
@@ -22,22 +17,16 @@ export function __kobold_fragment_unmount(f)
...
-	f.appendChild(e);
-	f.insertBefore(b, f.firstChild);
-}
-export function __kobold_fragment_drop(f)
-{
-	delete f.$begin;
-	delete f.$end;
+	f.appendChild(e).insertBefore(b, f.firstChild);
 }
+export function __kobold_fragment_drop(f) { delete f.$begin & delete f.$end; }
```
Comment (maciejhirsz) [inline crates/kobold/js/util.js:29]: This is fine but I also kind of hate it :sweat_smile:

File: crates/kobold/js/util.js [hunk 4]
```diff
@@ -22,22 +17,16 @@ export function __kobold_fragment_unmount(f)
...
 	while (b.nextSibling !== e) f.appendChild(b.nextSibling);
 	b.replaceWith(n);
-	f.appendChild(e);
-	f.insertBefore(b, f.firstChild);
-}
-export function __kobold_fragment_drop(f)
-{
-	delete f.$begin;
-	delete f.$end;
+	f.appendChild(e).insertBefore(b, f.firstChild);
```
Comment (maciejhirsz) [inline crates/kobold/js/util.js:27]: Ditto about `appendChild`

**Review by maciejhirsz — COMMENTED** [review]
I don't really think we need to minify anything here by hand. I'd rather have this code be really simple. If it needs to be minified it can be handled by `esbuild` or some such when you bundle all the little `.js` files together.

**Comment by maciejhirsz** [comment]
> before we can use it in an example main.rs file with `kobold::fragment()`?
> 
> but i can't see the purpose of `fragement`

There is a higher level RAII `Fragment` struct here:

https://github.com/maciejhirsz/kobold/blob/7fd98d2aec28c06b031efcae2e31405ca0683c9e/crates/kobold/src/dom.rs#L30-L51

All of those JS functions are just implementation details and shouldn't be exposed, the `html!` macro creates fragments just fine by default (no `<>...</>` needed).

File: crates/kobold/js/util.js (hunk 3 above)
Comment (ltfschoen) [reply crates/kobold/js/util.js:29]: yes i also prefer multi-line as it's more readable

File: crates/kobold/js/util.js (hunk 1 above)
Comment (ltfschoen) [reply crates/kobold/js/util.js:10]: thanks for the feedback! i've just been reading through chapter 4 of the rust book so i'm extra conscious of trying to remove any variables if their not necessary. i'll have a more detailed look at the code now.

**Comment by maciejhirsz** [comment]
Since there are no changes atm, I'll close it to keep the PRs clean.

**Pull request closed without merging**